### DISCUSSION TITLE
[Share] Update Arxiv Paper to latest version or published version

### DISCUSSION THREAD

**Discussion by seifer08ms** [windingwind/zotero-actions-tags#363] [Action Scripts]
### Description

The ArXiv entries created and stored in Zotero can be swiftly updated, with automatic detection of newer versions of ArXiv papers and identification of formally published ArXiv articles. This functionality can be readily implemented through the plugin [zotero-arxiv-workflow](https://github.com/AllanChain/zotero-arxiv-workflow). Recently, the author has released the relevant API( v0.2.2 ), which can be invoked by creating an Action to be integrated into the workflow, thereby automating the processing of our literature.

### Event

None

### Operation

Script

### Data

```javascript
/**
 * A description of this script.
 * @author Zotero Community
 * @usage 
 * @link https://github.com/windingwind/zotero-actions-tags/discussions/
 * @see https://github.com/windingwind/zotero-actions-tags/discussions/
 */

if (!item) return;
const Zotero = require("Zotero");
if (!Zotero.arXivWorkflow) {
	return "[Action:arXivWorkflow] arXivWorkflow for Zotero is not installed or disabled.";
}

if (item.itemType !== "preprint") return false;
await Zotero.arXivWorkflow.api.arXivUpdate(item);
```

### Anything else

This action can be incorporated into a "create item" process, where it is triggered by another action within a workflow (such as automatic translation, automatic tagging, or automatic metadata completion). This integration ensures that newly created papers identified as ArXiv preprints are automatically updated.
Thanks to the creators of Zotero-Actions-Tags, @windingwind, and Zotero-Arxiv-Workflow, @AllanChain, for providing these powerful plugin features.

**Comment by 3210448723** [comment]
老哥，能实现一个在zotero启动时更新所有的arxiv论文的脚本吗？

**Reply by seifer08ms** [reply]
最近好像api改了，我这边总出现问题，暂时恐怕不行。 而且我感觉插件目前可能也不太适合这么多论文一块更新，有空我再研究下。

**Reply by AllanChain** [reply]
> 最近好像api改了，我这边总出现问题，暂时恐怕不行。 而且我感觉插件目前可能也不太适合这么多论文一块更新，有空我再研究下。

BTW, in `0.3.0` you can pass an array of items to `Zotero.arXivWorkflow.api.arXivUpdate`.
